### PR TITLE
feat: respect acl when moving threads and mailboxes

### PR DESCRIPTION
**Pull request by kesselb**
Fix https://github.com/nextcloud/mail/issues/7909

The check for `i` does not cover all cases. 
To move `\SEEN`, `\DELETED` and `other flags` we need `s`, `t` and `w`.

### CODE REVIEW

File: src/util/acl.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	if (!mailbox.myAcls) {
+		return result
+	}
+
+	const acls = [...mailbox.myAcls]
+
+	for (const right of [...rights]) {
+		if (!acls.includes(right)) {
+			result = false
```
Comment (ChristophWurst) [inline src/util/acl.js:39]: you can probably return right away

Comment (kesselb) [reply src/util/acl.js:39]: good catch!

File: src/util/acl.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	const acls = [...mailbox.myAcls]
+
+	for (const right of [...rights]) {
+		if (!acls.includes(right)) {
+			return false
+		}
+	}
+
+	return true
+}
```
Comment (st3iny) [inline src/util/acl.js:42]: Epic! I wanted to apply something like that to the existing ACL code because there is a lot of code duplication.

**Comment by st3iny** [comment]
> To move \SEEN, \DELETED and other flags we need s, t and w

I think we can work on an optional base here. E.g. Thunderbird just ignores flags if necessary ACLs are missing and moves only the message.

**Review by st3iny — CHANGES_REQUESTED** [review]
Also need to check if the source envelope can be deleted (e and t on the source mailbox).

https://github.com/nextcloud/mail/blob/19cbdaee3c2e0d1dacda747b316293d63f149c85/src/components/Envelope.vue#L486-L505

File: src/util/acl.js (hunk 2 above)
Comment (GretaD) [reply src/util/acl.js:42]: My thoughts exactly! I can do it after all the issues are closed, in one big pr, if you dont mind

**Comment by kesselb** [comment]
> Also need to check if the source envelope can be deleted (e and t on the source mailbox).

True, but a different ticket ;)

This ticket is about "Filter target mailboxes for ACLs permissions in move modal". 
If the source mailbox or envelope lacks a permission we should hide the move action item. 

@hamza221 is already working on it in: https://github.com/nextcloud/mail/pull/8040

File: src/util/acl.js (hunk 2 above)
Comment (kesselb) [reply src/util/acl.js:42]: Thank you!

**Comment by st3iny** [comment]
> > Also need to check if the source envelope can be deleted (e and t on the source mailbox).
> 
> True, but a different ticket ;)
> 
> This ticket is about "Filter target mailboxes for ACLs permissions in move modal". If the source mailbox or envelope lacks a permission we should hide the move action item.
> 
> @hamza221 is already working on it in: #8040

Makes sense. All good then.

**Pull request merged**